src/goto-instrument/contracts/dynamic-frames/dfcc_dsl_contract_handler.cpp

```diff
@@ -0,0 +1,210 @@
+/*******************************************************************\
+
+Module: Dynamic frame condition checking for function contracts
+
+Author: Remi Delmas, delmasrd@amazon.com
+Date: August 2022
+
+\*******************************************************************/
+
+#include "dfcc_dsl_contract_handler.h"
+
+#include <util/arith_tools.h>
+#include <util/c_types.h>
+#include <util/expr_util.h>
+#include <util/format_expr.h>
+#include <util/format_type.h>
+#include <util/fresh_symbol.h>
+#include <util/invariant.h>
+#include <util/mathematical_expr.h>
+#include <util/namespace.h>
+#include <util/pointer_offset_size.h>
+#include <util/std_expr.h>
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/remove_function_pointers.h>
+
+#include <ansi-c/c_expr.h>
+#include <goto-instrument/contracts/contracts.h>
+#include <goto-instrument/contracts/utils.h>
+#include <langapi/language_util.h>
+
+#include "dfcc_dsl_wrapper_program.h"
+#include "dfcc_instrument.h"
+#include "dfcc_library.h"
+#include "dfcc_spec_functions.h"
+#include "dfcc_utils.h"
+
+std::map<irep_idt, dfcc_dsl_contract_functionst>
+  dfcc_dsl_contract_handlert::contract_cache;
```

Comment:
needs to be shared by possibly multiple instances of `dfcc_dsl_contract_handlert` to avoid generating/instrumenting a same contract more than once

Reply:
But ... the declaration in the cpp file is of no use?

Reply:
static class member have to be initialised out of line otherwise this happens:

```
Undefined symbols for architecture x86_64:
  "dfcc_dsl_contract_handlert::contract_cache", referenced from:
      dfcc_dsl_contract_handlert::get_contract_functions(dstringt const&) in libgoto-instrument-lib.a(dfcc_dsl_contract_handler.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[10/11] Linking CXX executable bin/unit
ninja: build stopped: subcommand failed.
```